httpbin/cmd/cmd_test.go

```diff
@@ -53,8 +53,9 @@ func TestLoadConfig(t *testing.T) {
 
 	testCases := map[string]struct {
 		args        []string
-		env         map[string]string
+		optionsEnv  map[string]string
 		getHostname func() (string, error)
+		env         []string
```

Comment:
This change adds a lot of unnecessary noise to the diff, I'd rather we take a different approach here.  Since we're already defining an env for these test cases, let's just use that env for both our `getEnvVal` and `getEnviron` stubs?

I.e. do something like this below:

```diff
- gotCode := mainImpl(tc.args, func(key string) string { return tc.optionsEnv[key] }, func() []string { return tc.env }, tc.getHostname, buf)
+ gotCode := mainImpl(tc.args, func(key string) string { return tc.env[key] }, func() []string { return environSlice(tc.env) }, tc.getHostname, buf)
```

where `environSlice` is a simple little helper to convert map[string]string into a `[]string` slice of key=val pairs.

Reply:
> It does look like we need a test for properly parsing the environment in mainImpl, I don't think I see any at a glance?

Yes, correct, I haven't managed to figure how to utilise the `mainImpl` for testing of the `/env` yet with some prefabricated `HTTPBIN_ENV_*` variables.
Perhaps, the `HTTPBIN_ENV_*` environ should become a member of the `config`, then the testing would become part of `loadConfig` test cases.
I'll welcome any suggestions. 

> This change adds a lot of unnecessary noise to the diff (...) where environSlice is a simple little helper (...)

I have reverted that `env` and `optionsEnv` splitting and implemented your suggestion.

Reply:
> Yes, correct, I haven't managed to figure how to utilise the `mainImpl` for testing of the `/env` yet with some prefabricated `HTTPBIN_ENV_*` variables.
> Perhaps, the `HTTPBIN_ENV_*` environ should become a member of the `config`, then the testing would become part of `loadConfig` test cases.

Ahhhhh yeah, I think it might be worth it to move this new env configuration onto the config struct, just to make it testable.  Would you mind making that change?  After that, I think we're good to go here!

Reply:
> Would you mind making that change?

Yes, of course, I will continue near the end of this week.

Reply:
@mccutchen As promised, I've just reworked this PR to store the env in the config and expose it via HTTPBin, and covered it with tests based on `loadConfig` test cases.

I don't mind to keep working on polishing this PR until you're happy to merge it, no rush with accepting it.